### PR TITLE
Feature/an 1580 fc sotre badges

### PR DESCRIPTION
**Pull request by Trinkes**

### CODE REVIEW

**Comment by jdandrade** [comment]
bumping constraint layout version broke the timeline stats/user bar in the store timeline context.

File: app/src/main/java/cm/aptoide/pt/store/view/GridStoreMetaWidget.java [hunk 1]
```diff
@@ -90,279 +94,389 @@ public GridStoreMetaWidget(View itemView) {
...
-    editStoreButton.setTextColor(color);
   }
 
-  private int getColorOrDefault(StoreTheme theme, Context context) {
-    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-      return context.getResources()
-          .getColor(theme.getPrimaryColor(), context.getTheme());
+  private void showMainIcon(String mainIconUrl) {
+    ImageLoader.with(getContext())
+        .loadWithShadowCircleTransform(mainIconUrl, mainIcon);
```
Comment (jdandrade) [inline app/src/main/java/cm/aptoide/pt/store/view/GridStoreMetaWidget.java]: Question: In a user has no store but has username and NO avatar, wouldn't this crash? Is this scenario even possible?

File: app/src/main/java/cm/aptoide/pt/view/app/ListStoreAppsFragment.java [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+
+/**
+ * Created by trinkes on 17/10/2017.
+ */
+
+public class ListStoreAppsFragment extends BackButtonFragment implements ListStoreAppsView {
+
+  public static final String STORE_ID = "cm.aptoide.pt.ListStoreAppsFragment.storeId";
+  public static final int LOAD_THRESHOLD = 5;
+  private static final String LIST_STATE_KEY = "cm.atpoide.pt.ListStoreAppsFragment.ListState";
```
Comment (jdandrade) [inline app/src/main/java/cm/aptoide/pt/view/app/ListStoreAppsFragment.java]: aptoide typo

File: build.gradle [hunk 3]
```diff
@@ -107,7 +107,7 @@ ext {
   LIB_PHONE_NUMBER_VERSION = '8.3.0'
   OKHTTP_VERSION = '3.6.0'
 
-  CONSTRAINT_LAYOUT_VERSION = '1.0.0'
+  CONSTRAINT_LAYOUT_VERSION = '1.0.2'
```
Comment (jdandrade) [inline build.gradle]: this breaks user timeline stats bar.

File: app/src/main/java/cm/aptoide/pt/store/view/GridStoreMetaDisplayable.java [hunk 4]
```diff
@@ -36,34 +41,164 @@ public GridStoreMetaDisplayable(GetHomeMeta pojo,
...
+    return accountManager.accountStatus()
+        .first()
+        .map(account -> {
+          if (getStore() == null || account.getStore() == null) {
+            return false;
+          }
+          return account.getStore()
+              .getName()
+              .equals(getStore().getName());
+        });
```
[the remark this answers is not in the document]
Comment (Trinkes) [reply app/src/main/java/cm/aptoide/pt/store/view/GridStoreMetaDisplayable.java]: fixed

File: app/src/main/java/cm/aptoide/pt/store/view/GridStoreMetaWidget.java [hunk 5]
```diff
@@ -16,70 +17,73 @@
...
+import rx.Observable;
 import rx.android.schedulers.AndroidSchedulers;
-import rx.functions.Action1;
 
 /**
  * Created by neuro on 04-08-2016.
  */
 public class GridStoreMetaWidget extends MetaStoresBaseWidget<GridStoreMetaDisplayable> {
 
+  private static final String TAG = GridStoreMetaWidget.class.getSimpleName();
```
[the remark this answers is not in the document]
Comment (Trinkes) [reply app/src/main/java/cm/aptoide/pt/store/view/GridStoreMetaWidget.java]: fixed

File: app/src/main/java/cm/aptoide/pt/store/view/GridStoreMetaWidget.java [hunk 6]
```diff
@@ -90,279 +94,389 @@ public GridStoreMetaWidget(View itemView) {
...
+    if (displayable.hasStore()) {
+      fragmentNavigator.navigateTo(
+          TimeLineFollowingFragment.newInstanceUsingStoreId(displayable.getStoreId(),
+              displayable.getStoreTheme()
+                  .getThemeName(), screenTitle, StoreContext.meta), true);
     } else {
-      ImageLoader.with(context)
-          .loadWithShadowCircleTransform(mainIconUrl, mainIcon);
+      fragmentNavigator.navigateTo(
+          TimeLineFollowingFragment.newInstanceUsingUserId(displayable.getUserId(), "DEFAULT",
```
[the remark this answers is not in the document]
Comment (Trinkes) [reply app/src/main/java/cm/aptoide/pt/store/view/GridStoreMetaWidget.java]: fixed

File: app/src/main/java/cm/aptoide/pt/store/view/GridStoreMetaWidget.java [hunk 7]
```diff
@@ -90,279 +94,389 @@ public GridStoreMetaWidget(View itemView) {
...
+        AptoideUtils.StringU.getFormattedString(R.string.social_timeline_followers_fragment_title,
+            resources, displayable.getFollowersCount());
+    if (displayable.hasStore()) {
+      fragmentNavigator.navigateTo(
+          TimeLineFollowersFragment.newInstanceUsingStore(displayable.getStoreId(),
+              displayable.getStoreTheme()
+                  .getThemeName(), screenTitle, StoreContext.meta), true);
+    } else {
+      fragmentNavigator.navigateTo(
+          TimeLineFollowersFragment.newInstanceUsingUser(displayable.getUserId(), "DEFAULT",
```
[the remark this answers is not in the document]
Comment (Trinkes) [reply app/src/main/java/cm/aptoide/pt/store/view/GridStoreMetaWidget.java]: fixed

File: app/src/main/java/cm/aptoide/pt/store/view/GridStoreMetaWidget.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Trinkes) [reply app/src/main/java/cm/aptoide/pt/store/view/GridStoreMetaWidget.java]: not possible

File: app/src/main/java/cm/aptoide/pt/store/view/my/MyStoreWidget.java [hunk 8]
```diff
@@ -118,13 +108,14 @@ public MyStoreWidget(View itemView) {
...
       return context.getResources()
-          .getColor(theme.getPrimaryColor(), context.getTheme());
+          .getColor(R.color.default_color, context.getTheme());
     } else {
       return context.getResources()
-          .getColor(theme.getPrimaryColor());
+          .getColor(R.color.default_color);
     }
   }
 }
```
[the remark this answers is not in the document]
Comment (Trinkes) [reply app/src/main/java/cm/aptoide/pt/store/view/my/MyStoreWidget.java]: fixed

File: app/src/main/java/cm/aptoide/pt/view/app/AlreadyLoadingException.java [hunk 9]
```diff
@@ -0,0 +1,8 @@
+package cm.aptoide.pt.view.app;
+
+/**
+ * Created by trinkes on 23/10/2017.
+ */
+
+public class AlreadyLoadingException extends RuntimeException {
+}
```
[the remark this answers is not in the document]
Comment (Trinkes) [reply app/src/main/java/cm/aptoide/pt/view/app/AlreadyLoadingException.java]: fixed

File: app/src/main/java/cm/aptoide/pt/view/app/Application.java [hunk 10]
```diff
@@ -0,0 +1,57 @@
+package cm.aptoide.pt.view.app;
+
+/**
+ * Created by trinkes on 18/10/2017.
+ */
+
+public class Application {
+  private final String name;
+  private final String icon;
+  private final float avg;
```
[the remark this answers is not in the document]
Comment (Trinkes) [reply app/src/main/java/cm/aptoide/pt/view/app/Application.java]: fixed

File: app/src/main/java/cm/aptoide/pt/view/app/ListStoreAppsFragment.java (hunk 2 above)
Comment (Trinkes) [reply app/src/main/java/cm/aptoide/pt/view/app/ListStoreAppsFragment.java]: fixed

File: build.gradle (hunk 3 above)
Comment (Trinkes) [reply build.gradle]: fixed

File: app/src/main/res/layout/store_badge_dialog.xml [hunk 11]
```diff
@@ -0,0 +1,194 @@
...
+        android:layout_marginTop="16dp"
+        android:gravity="center"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintRight_toRightOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/medal_icon"
+        tools:text="Gold medal"
+        style="@style/Aptoide.TextView.Medium.L"
+        />
```
Comment (jdandrade) [inline app/src/main/res/layout/store_badge_dialog.xml]: this textview does not have margins

**Pull request merged**